src/Simplex/Messaging/Protocol.hs

```diff
@@ -336,6 +340,8 @@ data BrokerMsg where
   MSG :: RcvMessage -> BrokerMsg
   NID :: NotifierId -> RcvNtfPublicDhKey -> BrokerMsg
   NMSG :: C.CbNonce -> EncNMsgMeta -> BrokerMsg
+  RKEY :: C.PublicKeyX25519 -> C.Signature C.Ed25519 -> BrokerMsg -- TLS-signed server key for proxy shared secret and initial sender key
+  RRES :: ByteString -> BrokerMsg -- Encrypted response. E.g. relay to client, hidden from proxy
```

Comment:
```suggestion
  RRES :: ByteString -> BrokerMsg -- Encrypted response. E.g. relay to client, hidden from proxy
  PRRES :: ByteString -> BrokerMsg -- Encrypted response. E.g. relay to client, hidden from proxy
```

I would use different newtypes for these bytestrings, with maybe max-length encoded in types, to make sure we correctly wrap/unwrap encryption layers.